lib/modules/onc_program/onc_standalone_launch_sequence.rb

```diff
@@ -109,6 +109,35 @@ def instance_scopes
         @instance.onc_sl_scopes
       end
 
+      def after_save_refresh_token(refresh_token)
+        # This method is used to save off the refresh token for standalone
+        # launch to be used for token revocation later.  We must do this because
+        # we are overwriting our standalone refresh/access token with the one
+        # used in the ehr launch.
+
+        # Note that this is also done after the token refresh within the 'Single
+        # Patient API' set of sequences, so this is somewhat redundant if that
+        # is implemented as expected.  However, we duplicate it here in case
+        # that isn't implemented as expected -- for example by failing that set
+        # of tests or if they interpretted the spec differently and do not echo
+        # back the 'refresh_token' parameter.
+
+        @instance.onc_sl_refresh_token = refresh_token
+        @instance.save!
+      end
+
+      def after_save_access_token(token)
+        # See `after_save_refresh_token` method for explanation of purpose of
+        # this method.
+
+        @instance.onc_sl_token = token
+
+        # save a copy so patient_id and oauth_token_endpoint are not overwritten
```

Comment:
I don't see `oauth_token_endpoint` being saved.

Reply:
This made me nervous, so I reviewed how we are saving fields 'off to the side'.  I found that we were a little sloppy with reporting when certain things were saved, and where they were saved, particularly from an 'Output' tab perspective.  I went through and cleaned it up all.  I think, when we wrote this all, we weren't expecting that prefilled data in the 'Token Refresh' tab would be relied on quite so heavily, so having it work was good enough.  But now it is clear that is a bit more than just convenience for the tester, because they get confused when it isn't filled.

I recommend looknig again from the start, since I've made a lot more changes.